subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchainQueryService.java

```diff
@@ -73,15 +80,23 @@ private JavaToolchain query(JavaToolchainSpec filter) {
         if (filter instanceof SpecificInstallationToolchainSpec) {
             return asToolchain(((SpecificInstallationToolchainSpec) filter).getJavaHome(), filter).get();
         }
-        return registry.listInstallations().stream()
+
+        Optional<JavaToolchain> currentJvmAsMatchingToolchain = Stream.of(asToolchain(currentJvmHome, filter))
```

Comment:
I am afraid this is too simplistic. The current JVM should only be considered if auto-detection is enabled. If it is not, then it should not be selected and only specific configured toolchain should be selected.